src/tree/VirtualMachineTreeItem.ts

```
@@ -86,49 +87,29 @@ export class VirtualMachineTreeItem extends AzureTreeItem {
+            // the message needs to constructed here since resourcesToDelete gets spliced
+            const deleteSucceeded: string = multiDelete ? localize('DeleteSucceeded', 'Successfully deleted {0}.', context.resourceList) :
+                localize('DeleteSucceeded', 'Successfully deleted {0} "{1}".', resourcesToDelete[0].resourceType, resourcesToDelete[0].resourceName);
 
-            const deleteSucceeded: string = multiDelete ? localize('DeleteSucceeded', 'Successfully deleted {0}.', resourceList) :
-                localize('DeleteSucceeded', 'Successfully deleted {0} "{1}".', resourcesToDelete[0].data.resourceType, resourcesToDelete[0].data.resourceName);
+            const errors: ResourceDeleteError[] = await deleteAllResources(this.root, this.resourceGroup, resourcesToDelete);
 
             const formattedErrors: string = errors.map(err => '\n' + parseError(err.error).message).join(',');
             const outputDeleteWithErrors: string = localize('outputDeleteWithErrors', `Failed to delete resources with the following errors: ${formattedErrors}`);
-            const messageDeleteWithErrors: string = localize('messageDeleteWithErrors', `Failed to delete the following resources ${errors.map(err => `"${err.resource.resourceName}"`).join(',')}. Check the output channel for more information.`);
+            const messageDeleteWithErrors: string = localize(
+                'messageDeleteWithErrors',
+                `Failed to delete the following resources ${errors.map(err => `"${err.resource.resourceName}"`).join(',')}. Check the output channel for more information.`);
 
-            ext.outputChannel.appendLog(errors.length > 0 ? outputDeleteWithErrors : deleteSucceeded);
+            if (multiDelete) { ext.outputChannel.appendLog(errors.length > 0 ? outputDeleteWithErrors : deleteSucceeded); }
             vscode.window.showInformationMessage(errors.length > 0 ? messageDeleteWithErrors : deleteSucceeded);
```

In the error case it should use `showErrorMessage` and we probably want to throw an error so telemetry will treat this as a failure